codec-http2/src/main/java/io/netty/handler/codec/http2/Http2LocalFlowController.java

```diff
@@ -66,4 +66,50 @@ void receiveFlowControlledFrame(ChannelHandlerContext ctx, Http2Stream stream, B
      * @return the number of unconsumed bytes for the stream.
      */
     int unconsumedBytes(Http2Stream stream);
+
+    /**
+     * Factory method to generate a new {@link LocalFlowState} object. This will be called after a new stream object
+     * is created and before {@link Http2Connection.Listener#onStreamAdded(Http2Stream)} is called. The result of this
+     * method will be returned by {@link Http2Stream#localFlowState()} for use by {@link Http2LocalFlowController}.
+     * @param stream The stream to associate the {@link LocalFlowState} with.
+     * This may be any stream, including the {@link Http2Connection#connectionStream()}.
+     * @return A new object which will represent local flow control state for {@code stream}.
+     */
+    LocalFlowState newLocalFlowState(Http2Stream stream);
```

Comment:
@nmittler - By sending a window update I meant adjusting the initial window size of a stream.  This was meant as an example of something that may be useful and not the only thing that can be done....as resources change this may lead to interesting use cases to act on.  I'm not sure extending the flow controllers provide this visibility.  We are intentionally hiding the `DefaultFlowState` implementations which provide most of the logic.  It also seems like providing a new implementation of flow control to get this state may be overkill. This type of information is less of a extension or new implementation but just seems like general state associated with the flow control algorithm described in the spec.